tap_kit/streams.py

```diff
@@ -148,3 +152,21 @@ def update_start_date_bookmark(self):
     def update_and_return_bookmark(self):
         self.update_start_date_bookmark()
         return self.get_bookmark()
+
+def validate_ingestible_data(record):
+    from pycore.text import to_unicode
+    import re
+    
+    for key, value in record.items():
+        if isinstance(value, dict):
+            validate_ingestible_data(value)
+        else:
+            try:
+                record[key] = to_unicode(value)
+            except UnicodeDecodeError as e:
+                record[key] = ''
+                pass
+            if len(str(value)) != len(str(value).encode()):
+                record[key] = value.encode("ascii", "ignore").replace(b'\x00', b'').decode()
```

Comment:
if we're just trying to remove non utf-8 chars can't we encode to utf-8, ignore errors, then decode?  then we can skip the replace part.

Reply:
NUL is a valid uf-8 value.  Is this causing an issue downstream?

Reply:
Joel and I discussed and he found CSVs just do not support nul bytes so cleaned up the code to only strip out nulls